src/screens/MessageScreen.tsx

```diff
@@ -136,10 +153,51 @@ const defaultStyles = createStyles('MessageScreen', (theme) => {
     },
     loadMoreButton: {},
     loadMoreText: {},
+    listItemTimeText: { paddingLeft: 15 },
+    newMessageIconView: {
+      marginRight: -1,
+      borderWidth: 2,
+      borderColor: theme.colors.text,
+      borderRadius: 32,
+    },
+    newMessageText: {
+      color: theme.colors.text,
+      fontWeight: '600',
+    },
   };
 });
 
 declare module '@styles' {
   interface ComponentStyles
     extends ComponentNamedStyles<typeof defaultStyles> {}
 }
+
+const formatMessageTime = (timestamp?: string) => {
+  if (!timestamp) {
+    return '';
+  }
+
+  const currentDate = new Date();
+  const date = new Date(timestamp);
+
+  const secondsDifference = differenceInSeconds(currentDate, date);
+  const minutesDifference = differenceInMinutes(currentDate, date);
+  const differenceHours = differenceInHours(currentDate, date);
+  const daysDifference = differenceInDays(currentDate, date);
+
+  if (secondsDifference < 60) {
+    return 'now';
+  } else if (minutesDifference < 60) {
+    return `${minutesDifference}min`;
+  } else if (differenceHours === 1) {
+    return `${differenceHours}hour`;
+  } else if (differenceHours < 24) {
+    return `${differenceHours}hrs`;
+  } else if (daysDifference === 1) {
+    return `${daysDifference}day`;
+  } else if (daysDifference < 3) {
+    return `${daysDifference}days`;
+  } else {
+    return format(date, 'MMM d');
```

Comment:
I think most time libraries support this kind of thing by default. I think for `date-fns` we'd want https://date-fns.org/v2.30.0/docs/formatDistance. We could get the shorter text by using the custom formatting letters.

Reply:
I tried to use formatDistance but it doesn't support the custom letter from what I could tell. I'll definitely revisit this in the future and put something more standard in place but I think it's fine for now.

Reply:
what is the custom letter we need? the `min/hrs/hour` etc?

Reply:
not sure what this suffix does exactly but I saw it quickly

```
formatDistance(subDays(new Date(), 3), new Date(), { addSuffix: true })
```